source/common/network/proxy_protocol_filter_state.cc

```diff
@@ -9,5 +12,66 @@ const std::string& ProxyProtocolFilterState::key() {
   CONSTRUCT_ON_FIRST_USE(std::string, "envoy.network.proxy_protocol_options");
 }
 
+class ProxyProtocolFilterStateReflection : public StreamInfo::FilterState::ObjectReflection {
+public:
+  ProxyProtocolFilterStateReflection(const ProxyProtocolFilterState* object) : object_(object) {}
+
+  FieldType getField(absl::string_view tlv_type_str) const override {    
+    // Specified tlv_type must be parsable as an int.
+    ASSERT(!tlv_type_str.empty());
+    int tlv_type;
+    if (!absl::SimpleAtoi(tlv_type_str, &tlv_type)) {
+    throw EnvoyException(fmt::format(
+        "Invalid parameter provided for FIELD value: {}. The proxy protocol TLV type must be parsable as int.",
+        tlv_type_str));
+    }
+
+    // Check if a valid TLV type was passed in.
+    if (tlv_type >= 256 || tlv_type <= 0) {
+      throw EnvoyException(fmt::format("Invalid parameter provided for FIELD value: "
+                                      "{}. The proxy protocol TLV type must be a positive integer less than 256.",
+                                      tlv_type_str));
+    }
```

Comment:
I think you shouldn't throw exception here. As I said in previous PR, the only drawback of the reflection design is we cannot validate the input parameter for now.

Because the `getField` will be called at runtime and the exception will crash Envoy. It would be better to return `absl::monostate` directly.